crates/ntx-builder/src/state/mod.rs

```diff
@@ -27,12 +28,25 @@ pub struct TransactionCandidate {
     pub account: Account,
     /// A set of notes addressed to this network account.
     pub notes: Vec<NetworkNote>,
+    /// The latest locally committed block header.
+    ///
+    /// This should be used as the reference block during transaction execution. It is guaranteed
+    /// to be tracked by `chain_mmr`.
+    pub chain_tip: BlockHeader,
```

Comment:
One thing worth clarifying si whether `chain_tip` is already in the `chain_mmr` or not (I think it isn't).